src/main/java/com/whalespottingjava/services/PostCodeService.java

```diff
@@ -0,0 +1,78 @@
+package com.whalespottingjava.services;
+
+import java.io.IOException;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+import org.json.JSONObject;
+
+public class PostCodeService {
+    private String urlPostCodeApi = "https://api.postcodes.io/postcodes";
+    private String resultKeyWord = "result";
+    private String postCodeKeyWord = "postcode";
+    private String httpRequestPostCodeApi(String url) throws URISyntaxException {
+        HttpRequest request = HttpRequest.newBuilder().uri(new URI(url))
+                .version(HttpClient.Version.HTTP_2).GET().build();
+        String responseBody;
+        try {
+            HttpResponse<String> response = HttpClient.newBuilder().build()
```

Comment:
This should be declared at the top of the method and instantiated in the try...catch. General rules for variable declaration:

- Declare everything at the top
- In order of complexity

Reply:
thanks!